juju_verify/verifiers/base.py

```diff
@@ -165,9 +172,12 @@ def check_affected_machines(self) -> Result:
             ):
                 machine_map[unit.machine.entity_id].append(unit.entity_id)
 
+        verified_units = self.unit_ids.copy()
+        verified_units.extend(unit.entity_id for unit in self.exclude_affected_units)
+
         for machine, unit_list in machine_map.items():
             for unit in unit_list:
-                if unit not in self.unit_ids:
+                if unit not in verified_units:
```

Comment:
I think it will be much cleaner this way `unit not in self.unit_ids and unit not in  self.exclude_affected_units`.

Reply:
This is probably a matter of preference. I prefer the condition to be shorter. Maybe @auria can weigh in here, which one should be used.

Reply:
This is related to previous comment about using `List[str]` or `Set[str]` for `self.exclude_affected_units`, where `str` is `entity_id`.